app/pages/sections/catalogue-solutions/order-item/controller.js

```diff
@@ -22,16 +22,56 @@ export const getSelectedPrice = async ({ selectedPriceId, accessToken }) => {
 
 export const getOrderItemContext = async ({
   orderId,
-  selectedSolutionId,
+  solutionName,
   selectedRecipientId,
-  selectedPriceId,
-  accessToken,
-}) => {
-  const solutionName = (await getSolution({ solutionId: selectedSolutionId, accessToken })).name;
-  const serviceRecipientName = await getRecipientName({ selectedRecipientId, accessToken });
-  const selectedPrice = await getSelectedPrice({ selectedPriceId, accessToken });
-
-  return getContext({
-    orderId, solutionName, serviceRecipientName, odsCode: selectedRecipientId, selectedPrice,
-  });
+  serviceRecipientName,
+  selectedPrice,
+}) => getContext({
+  orderId, solutionName, serviceRecipientName, odsCode: selectedRecipientId, selectedPrice,
+});
+
+export const getOrderItemErrorPageContext = params => getErrorContext(params);
+
+export const validateOrderItemForm = ({ data }) => {
+  const errors = [];
+  if (!data.quantity || data.quantity.trim().length <= 0) {
+    errors.push({
+      field: 'quantity',
+      id: 'quantityRequired',
+    });
+  }
+
+  if (!data.quantity || isNaN(data.quantity)) {
+    errors.push({
+      field: 'quantity',
+      id: 'numericQuantityRequired',
+    });
+  }
```

Comment:
is this right? If I put in `blah` into quantity I am going to get Quantity is required rather that Quantity must be a number

Reply:
ignore me..  I missed the first bit.

Reply:
<img width="832" alt="image" src="https://user-images.githubusercontent.com/30870498/86116257-07c69580-bac5-11ea-87d7-82f23c13effb.png">